File: Backend/cpp/kisi_islemleri.cpp
#include <iostream>
#include <pqxx/pqxx>

int main() {
    try {
        pqxx::connection conn("host=db user=kullanici password=sifre dbname=rehber");
        pqxx::work txn(conn);
        txn.exec(
            "INSERT INTO kisi (isim, eposta, telefon, adres) "
            "VALUES ('Mehmet Demir', 'mehmet@example.com', '5559876543', 'İstanbul, Türkiye')"
        );
        txn.commit();
        std::cout << "Kişi eklendi." << std::endl;
    } catch (const std::exception &e) {
        std::cerr << "Hata: " << e.what() << std::endl;
        return 1;
    }
    return 0;
}

File: Backend/csharp/KisiIslemleri.cs
using System;
using Npgsql;

public class KisiIslemleri
{
    private static string connectionString = "Host=db;Username=kullanici;Password=sifre;Database=rehber";

    public static void KisiEkle(string isim, string eposta, string telefon, string adres)
    {
        using (var connection = new NpgsqlConnection(connectionString))
        {
            connection.Open();
            var command = new NpgsqlCommand(
                "INSERT INTO kisi (isim, eposta, telefon, adres) VALUES (@isim, @eposta, @telefon, @adres)",
                connection
            );
            command.Parameters.AddWithValue("@isim", isim);
            command.Parameters.AddWithValue("@eposta", eposta);
            command.Parameters.AddWithValue("@telefon", telefon);
            command.Parameters.AddWithValue("@adres", adres);
            command.ExecuteNonQuery();
        }
    }
}
